File: JSGame/game.js
/**
 * Arojen-tarinat - JavaScript version
 * A simple console-based strategy game
 */

const { factions } = require('./Factions');

function main() {
    console.log("Tervetuloa Arojen-tarinoihin!");
    console.log("Valitse heimosi:");
    factions.forEach((faction, i) => {
        console.log(`${i + 1}. ${faction.name} - ${faction.bonus}`);
    });

    const readline = require('readline');
    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    rl.question("Valinta (1-4): ", (answer) => {
        const choice = parseInt(answer) - 1;
        const playerFaction = factions[choice];
        console.log(`Valitsit: ${playerFaction.name}`);
        console.log(`Aloitusyksiköt: ${playerFaction.startUnits.join(', ')}`);

        // Simple game loop
        let turn = 1;
        function gameLoop() {
            console.log(`\nVuoro ${turn}`);
            console.log("1. Liiku");
            console.log("2. Taistele");
            console.log("3. Lopeta");
            rl.question("Toiminto: ", (action) => {
                if (action === "3") {
                    console.log("Peli päättyi!");
                    rl.close();
                    return;
                } else if (action === "1") {
                    console.log("Liikut yksikköä");
                } else if (action === "2") {
                    console.log("Taistelet");
                }
                turn++;
                gameLoop();
            });
        }
        gameLoop();
    });
}

main();

File: PythonGame/game.py
"""
Arojen-tarinat - Python version
A simple text-based strategy game
"""

from Factions import factions

def main():
    print("Tervetuloa Arojen-tarinoihin!")
    print("Valitse heimosi:")
    for i, faction in enumerate(factions, 1):
        print(f"{i}. {faction['name']} - {faction['bonus']}")

    choice = int(input("Valinta (1-4): ")) - 1
    player_faction = factions[choice]
    print(f"Valitsit: {player_faction['name']}")
    print(f"Aloitusyksiköt: {', '.join(player_faction['startUnits'])}")

    # Simple game loop
    turn = 1
    while True:
        print(f"\nVuoro {turn}")
        print("1. Liiku")
        print("2. Taistele")
        print("3. Lopeta")
        action = input("Toiminto: ")
        if action == "3":
            break
        elif action == "1":
            print("Liikut yksikköä")
        elif action == "2":
            print("Taistelet")
        turn += 1

    print("Peli päättyi!")

if __name__ == "__main__":
    main()